binaryop/funcs.go

```diff
@@ -107,3 +107,21 @@ func Ifnot(left, right float64) float64 {
 	}
 	return nan
 }
+
+// And return left if left and right is not NaN. Otherwise, NaN is returned.
+func And(left, right float64) float64 {
```

Comment:
Could you verify other places in metricsql package and in the VictoriaMetrics package, which use the `binaryop.*` functions? Probably, they also need to be updated with the newly added `And` and `Or` functions?